tools/code_format/config.yaml

```diff
@@ -0,0 +1,297 @@
+    - test/integration/load_balancers/
+
+  # Files in these paths can use MessageLite::SerializeAsString
+  serialize_as_string:
+    include:
+    - source/common/protobuf/utility.cc
+    - source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc
+    - test/common/grpc/codec_fuzz_test.cc
+    - test/common/grpc/codec_test.cc
+    - test/common/protobuf/utility_test.cc
+    - test/extensions/bootstrap/wasm/test_data/speed_cpp.cc
+    - test/extensions/filters/common/expr/context_test.cc
+    - test/extensions/filters/http/common/fuzz/uber_filter.h
+    - test/tools/router_check/router_check.cc
+
+  # Files in these paths can use std::regex
+  std_regex:
+    include:
+    - contrib/squash/filters/http/source/squash_filter.h
+    - contrib/squash/filters/http/source/squash_filter.cc
+    - source/common/common/regex.h
+    - source/common/common/regex.cc
+    - source/common/common/utility.cc
+    - source/common/formatter/substitution_formatter.cc
+    - source/common/stats/tag_extractor_impl.h
+    - source/common/stats/tag_extractor_impl.cc
+    - source/common/version/version.cc
+    - source/server/admin/prometheus_stats.h
+    - source/server/admin/prometheus_stats.cc
+    - source/server/admin/stats_params.h
+    - source/server/admin/stats_request.cc
+    - source/server/admin/utils.h
+    - source/server/admin/utils.cc
+    - tools/clang_tools/api_booster/main.cc
+    - tools/clang_tools/api_booster/proto_cxx_utils.cc
+
+  # These are entire files that are allowed to use std::string_view vs. individual exclusions. Right
+  # now this is just WASM which makes use of std::string_view heavily so we need to convert to
+  # absl::string_view internally. Everywhere else should be using absl::string_view for additional
+  # safety.
+  std_string_view:
+    include:
+    - source/extensions/common/wasm/context.h
+    - source/extensions/common/wasm/context.cc
+    - source/extensions/common/wasm/foreign.cc
+    - source/extensions/common/wasm/wasm.h
+    - source/extensions/common/wasm/wasm.cc
+    - source/extensions/common/wasm/wasm_vm.h
+    - source/extensions/common/wasm/wasm_vm.cc
+    - test/extensions/bootstrap/wasm/wasm_speed_test.cc
+    - test/extensions/bootstrap/wasm/wasm_test.cc
+    - test/extensions/common/wasm/wasm_test.cc
+    - test/extensions/stats_sinks/wasm/wasm_stat_sink_test.cc
+    - test/test_common/wasm_base.h
+
+dir_order:
+  - envoy
+  - common
+  - source
+  - exe
+  - server
+  - extensions
+  - test
+
+re:
+  codeowners_contrib: (/contrib/[^@]*\s+)(@.*)
+  codeowners_extensions: .*(extensions[^@]*\s+)(@.*)
+  comment: //|\*
+  designated_initializer: \{\s*\.\w+\s*\=
+  # Check for punctuation in a terminal ref clause, e.g.
+  # :ref:`panic mode. <arch_overview_load_balancing_panic_threshold>`
+  dot_multi_space: \. +
+  duration_value: \b[Dd]uration\(([0-9.]+)
+  # C++17 feature, lacks sufficient support across various libraries / compilers.
+  for_each_n: for_each_n\(
```

Comment:
I'm not sure how I feel about having every constant moved out of the Python file into YAML. I certainly agree all the includes/excludes and things we want to keep changing belong there. Let me get a second opinion from @jmarantz, I don't feel that strongly.

Reply:
> I'm not sure how I feel about having every constant moved out of the Python file into YAML.

apart from making the script and data/config a lot clearer (separately) this allows the script to move to pytooling while the data/config it uses can remain in envoy itself where it can be seen/adjusted/etc by dev without having to worry about how eg substitutions are actually substituted